src/content/peerconnection/trickle-ice/js/main.js

```diff
@@ -164,7 +164,33 @@ function iceCallback(event) {
     appendCell(row, c.port);
     appendCell(row, formatPriority(c.priority));
   } else {
-    appendCell(row, 'Done', 7);
+    var result = 'Done';
+    if (servers.length === 1) {
+      var server = JSON.parse(servers[0].value);
+
+      // figure out the types of candidates (host, srflx, relay)
+      var types = pc.localDescription.sdp.split('\r\n').filter(
```

Comment:
This is pretty gnarly. I recommend breaking this out into its own getFinalResult function, as well as storing the candidate objects in a global list so that you don't have to process SDP.